### PR TITLE
User import implemented, Dataloader removed

### PR DESCRIPTION
**Pull request by Decitizen**
__Added functionality for importing User after login__
* New service `UserService`-class + JUnit tests
* New `OAuthUser`-class supports OAuth2 authentication 
* Implementation of `/user` for both `UserDevController` and `UserController` 

__Removed DataLoader__:
* `UserType`-enum updated and moved from `DataLoader` to `DataImporter`
* `DataLoader` removed, superseded by `DataImporter`

### CODE REVIEW

**Comment by Decitizen** [comment]
Controllers have been implemented and tested by hand, however they still need proper `MockMvc`-tests. I still need to figure what is the best practice for this (involves better understanding of the brand new Spring Security release).

**Comment by DeepIntuition** [comment]
__Don't merge this yet__, let's take @lramirez's PR first because its functionality is of higher priority atm and is conflicting with this one. I'll update my PR in order to accommodate the forthcoming changes.

**Comment by Decitizen** [comment]
* __Update:__ testing with MockMVC and Spring Security requires a renaming of class __User__ to __TMCUser__ in order to avoid ambiguity. However, this will yield changes in all of the classes that use `User` atm. I will introduce a completely new PR about the MockMVC tests. __For now this can be merged.__

File: src/test/java/pingis/controllers/UserControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    
+    @AfterClass
+    public static void tearDownClass() {
+    }
+    
+    /*
+    @Before
+    public void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(context).build();
+    }*/
```
Comment (laurpulk) [inline src/test/java/pingis/controllers/UserControllerTest.java]: /* */

**Comment by Decitizen** [comment]
Updated @dwarfcrank

**Pull request merged**